### PR TITLE
ZKVM-483: Enable Metal by default on macOS

### PR DESCRIPTION
**Pull request by flaub**

### CODE REVIEW

File: risc0/zkp/build.rs [hunk 1]
```diff
@@ -15,9 +15,11 @@
...
 
 fn main() {
-    if env::var("CARGO_FEATURE_METAL").is_ok() {
-        let metal_bin = env::var("DEP_RISC0_SYS_METAL_KERNELS_ZKP")
-            .expect("CARGO_FEATURE_METAL is defined, but DEP_RISC0_SYS_METAL_KERNELS_ZKP is not");
+    if env::var("CARGO_FEATURE_PROVE").is_ok()
+        && env::var("CARGO_CFG_TARGET_OS").is_ok_and(|os| os == "macos" || os == "ios")
+    {
+        let metal_bin =
+            env::var("DEP_RISC0_SYS_METAL_KERNELS_ZKP").expect("DEP_RISC0_SYS_METAL_KERNELS_ZKP");
```
Comment (SchmErik) [inline risc0/zkp/build.rs:22]: Perhaps something like this:
```suggestion
            env::var("DEP_RISC0_SYS_METAL_KERNELS_ZKP").expect("attempting to build risc0-zkp using the `prove` feature flag. This requires building the metal kernels but DEP_RISC0_SYS_METAL_KERNELS_ZKP is not defined.");
```

**Review by SchmErik — APPROVED** [review]
Looks good to me - I had a nit so it would be nice if you could address that but looks good other than that. Looking forward, I'm thinking that all of these
```
#[cfg(any(target_os = "macos", target_os = "ios"))]
pub mod metal;
```
Will change to something like
```
#[cfg(any(target_os = "macos", target_os = "ios", target_arch = "arm64"))]
pub mod metal;
```
and compile to CPU if the host is x86 mac

File: risc0/zkp/Cargo.toml [hunk 2]
```diff
@@ -52,9 +54,10 @@ rustdoc-args = ["--cfg", "docsrs"]
 features = ["prove", "std"]
 
 [features]
-default = []
+circuit_debug = []
 cuda = ["dep:cust", "prove", "risc0-sys/cuda"]
-metal = ["dep:metal", "prove", "risc0-sys/metal"]
+default = []
+metal_prefix_products = []
```
[the remark this answers is not in the document]
Comment (flaub) [reply risc0/zkp/Cargo.toml:60]: Yes, it's a partial implementation, but when it's complete it should give us a pretty good speed up.

**Comment by flaub** [comment]
I plan to support x86 in a followup. One issue is that we don't have CI to test that it works.

**Pull request merged**